framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -241,189 +265,234 @@ public int runTests() throws Exception {
-        LOG.info("Executed queries from " + queryFile);
-      }
-      if(cmdParam.runFailed == true){
-        if(passingTests.size()>0){
-          LOG.info(LINE_BREAK);
-      	  LOG.info("Passing Tests:");
-      	}
-      	for (DrillTest test : passingTests) {
-          LOG.info(test.getInputFile());
-      	}
+
+      if(executionFailures.size()>0 || dataVerificationFailures.size()>0 || planVerificationFailures.size()>0 || timeoutFailures.size()>0 || randomFailures.size() > 0) {
+        LOG.info("\n"+LINE_BREAK);
+        LOG.info("ITERATION RESULTS");
+        LOG.info(LINE_BREAK);
       }
+
       if(executionFailures.size()>0){
-        LOG.info(LINE_BREAK);
       	LOG.info("Execution Failures:");
         for (DrillTest test : executionFailures) {
-	  if(test.getExpectedFile()!=""){
+	      if(test.getExpectedFile()!=""){
             LOG.info(test.getInputFile());
-	  }
-      	}
-	if(cmdParam.runFailed == true){
-      	  LOG.info("Execution Failures with name errors:");
-      	}
-	for (DrillTest test : executionFailures) {
-          if(test.getExpectedFile()==""){
-	    LOG.info(test.getInputFile());
-	  }
+	      }
+        LOG.info("\n");
       	}
       }
+
       if(dataVerificationFailures.size()>0){
-        LOG.info(LINE_BREAK);
       	LOG.info("Data Verification Failures:");
+        for (DrillTest test : dataVerificationFailures) {
+          LOG.info(test.getInputFile());
+        }
+        LOG.info("\n");
       }
-      for (DrillTest test : dataVerificationFailures) {
-        LOG.info(test.getInputFile());
-      }
+
       if(planVerificationFailures.size()>0){
-        LOG.info(LINE_BREAK);
       	LOG.info("Plan Verification Failures:");
+        for (DrillTest test : planVerificationFailures) {
+          LOG.info(test.getInputFile());
+        }
+        LOG.info("\n");
       }
-      for (DrillTest test : planVerificationFailures) {
-        LOG.info(test.getInputFile());
-      }
+
       if(timeoutFailures.size()>0){
-      	LOG.info(LINE_BREAK);
-      	LOG.info("Timeout Failures:");
-      	LOG.info(LINE_BREAK);
-      }
-      for (DrillTest test : timeoutFailures) {
-        LOG.info(test.getInputFile());
+      	LOG.info("Timeouts:");
+        for (DrillTest test : timeoutFailures) {
+          LOG.info(test.getInputFile());
+        }
+        LOG.info("\n");
       }
+
       if(randomFailures.size()>0){
-      	LOG.info(LINE_BREAK);
       	LOG.info("Random Failures:");
-      	LOG.info(LINE_BREAK);
-      }
-      for (DrillTest test : randomFailures) {
+        for (DrillTest test : randomFailures) {
         LOG.info(test.getInputFile());
+        }
+        LOG.info("\n");
       }
+
+      // Special handling for repeated runs
+      if ((cmdParam.repeatRun == true) && (queryFile.length() > 0)) {
+        LOG.info("Executed queries from " + queryFile);
+      }
+
+      // Special handling for running tests tagged as failed
```

Comment:
Just moved these snippets below their previous position to clarify flow and improve readability.